src/alchemlyb/parsing/parquet.py

```diff
@@ -1,9 +1,42 @@
 import pandas as pd
+from loguru import logger
 
 from . import _init_attrs
 
 
-@_init_attrs
+def _check_metadata(path: str, T: float) -> pd.DataFrame:
```

Comment:
The name is a bit misleading because it does not only check but also return a df. I would choose a name such as `_read_parquet_with_metadata` or similar.